# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/ipc/BrowserChild.cpp
# Commit: c49115748677
# Full Hash: c491157486775555a61d748ccbbc054988aacae9
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2020-03-23 21:38:56
# Description:
#   Bug 1614259 - Ensure BrowisngContexts are detached when nsDocShell is destroyed, r=farre
#   
#   After bug 1582832, DocShell destruction and BrowsingContext detaching happen
#   in separate operations, leaving a gap where a DocShell has been destroyed, but
#   its BrowsingContext is still considered attached. During this gap, the usual
# ==============================================================================

diff -r e8f6b2875121 -r c49115748677 dom/ipc/BrowserChild.cpp
--- a/dom/ipc/BrowserChild.cpp	Mon Mar 23 15:19:14 2020 +0000
+++ b/dom/ipc/BrowserChild.cpp	Mon Mar 23 13:28:09 2020 +0000
@@ -1036,9 +1036,8 @@
 
 mozilla::ipc::IPCResult BrowserChild::RecvWillChangeProcess(
     WillChangeProcessResolver&& aResolve) {
-  if (nsCOMPtr<nsIDocShell> docShell = do_GetInterface(WebNavigation())) {
-    RefPtr<nsDocShell> docshell = nsDocShell::Cast(docShell);
-    docshell->SetWillChangeProcess();
+  if (mWebBrowser) {
+    mWebBrowser->SetWillChangeProcess();
   }
   aResolve(true);
   return IPC_OK();